roles/common/tasks/monitoring.yml

```diff
@@ -58,12 +58,65 @@
   lineinfile: dest=/etc/default/sensu regexp=^EMBEDDED_RUBY
               line=EMBEDDED_RUBY=true
   notify: restart sensu-client
+  
+- block:
+  - name: sensu python venv directory
+    file: dest="{{ common.python.sensu_venv }}"
+          state=directory mode=0755
+        
+  - name: set pip proxy args
+    set_fact:
+      pip_extra_args: " -i {{ openstack.pypi_mirror }}"
+    when: openstack.pypi_mirror is defined
+
+  - name: update pip
+    pip: name=pip version={{ common.pip_version }}
+         extra_args="-i {{ openstack.pypi_mirror }}"
+         virtualenv="{{ common.python.sensu_venv }}"
+    register: result
+    until: result|succeeded
+    retries: 5
+
+  - name: upgrade setuptools for sensu venv (osp)
+    pip:
+      name: "setuptools"
+      virtualenv: "{{ common.python.sensu_venv }}"
+      extra_args: "{{ pip_extra_args }} --upgrade"
+    register: result
+    until: result|succeeded
+    retries: 5
+
+  - name: install shade bits for ansible modules
+    pip:
+      name: "{{ item }}"
+      virtualenv: "{{ common.python.sensu_venv }}"
+    with_items:
+      - shade>=1.9.0
+      - python-dateutil
+    register: result
+    until: result|succeeded
+    retries: 5
+
+  - name: create symbolic link for ca cert
+    file: 
+      src: "{{ ca_bundle }}"
+      dest: "{{ common.python.sensu_venv }}/lib/python2.7/site-packages/requests/cacert.pem"
+      state: link
+      force: true
+
+  when: openstack_install_method == "distro"
```

Comment:
I think it would be a good idea to have sensu use a venv on all OS's. You could remove the block/conditional for 'distro' install method.